src/Evolution/Systems/Cce/BoundaryData.hpp

```diff
@@ -539,4 +539,620 @@ void du_j_worldtube_data(
+      ::Tags::deriv<Tags::detail::DLambda<Tags::detail::RealBondiR>,
+                    tmpl::size_t<2>, Frame::RadialNull>>>
+      computation_variables{size};
+
+  Variables<
+      tmpl::list<::Tags::SpinWeighted<::Tags::TempScalar<0, ComplexDataVector>,
+                                      std::integral_constant<int, 0>>,
+                 ::Tags::SpinWeighted<::Tags::TempScalar<0, ComplexDataVector>,
+                                      std::integral_constant<int, 1>>>>
+      derivative_buffers{size};
+
+  auto& cos_phi = get<Tags::detail::CosPhi>(computation_variables);
+  auto& cos_theta = get<Tags::detail::CosTheta>(computation_variables);
+  auto& sin_phi = get<Tags::detail::SinPhi>(computation_variables);
+  auto& sin_theta = get<Tags::detail::SinTheta>(computation_variables);
+  trigonometric_functions_on_swsh_collocation(
+      make_not_null(&cos_phi), make_not_null(&cos_theta),
+      make_not_null(&sin_phi), make_not_null(&sin_theta), l_max);
+
+  // NOTE: to handle the singular values of polar coordinates, the phi
+  // components of all tensors are scaled according to their sin(theta)
+  // prefactors.
+  // so, any down-index component get<2>(A) represents 1/sin(theta) A_\phi,
+  // and any up-index component get<2>(A) represents sin(theta) A^\phi.
+  // This holds for Jacobians, and so direct application of the Jacobians
+  // brings the factors through.
+  auto& cartesian_coords =
+      get<Tags::detail::CartesianCoordinates>(computation_variables);
+  auto& cartesian_to_spherical_jacobian =
+      get<Tags::detail::CartesianToSphericalJacobian>(computation_variables);
+  auto& inverse_cartesian_to_spherical_jacobian =
+      get<Tags::detail::InverseCartesianToSphericalJacobian>(
+          computation_variables);
+  cartesian_to_spherical_coordinates_and_jacobians(
+      make_not_null(&cartesian_coords),
+      make_not_null(&cartesian_to_spherical_jacobian),
+      make_not_null(&inverse_cartesian_to_spherical_jacobian), cos_phi,
+      cos_theta, sin_phi, sin_theta, extraction_radius);
+
+  auto& spatial_metric =
+      get<gr::Tags::SpatialMetric<3, ::Frame::Inertial, DataVector>>(
+          computation_variables);
+  gr::spatial_metric(make_not_null(&spatial_metric), spacetime_metric);
+
+  auto& inverse_spatial_metric =
+      get<gr::Tags::InverseSpatialMetric<3, ::Frame::Inertial, DataVector>>(
+          computation_variables);
+  inverse_spatial_metric = determinant_and_inverse(spatial_metric).second;
+
+  auto& shift = get<gr::Tags::Shift<3, ::Frame::Inertial, DataVector>>(
+      computation_variables);
+  gr::shift(make_not_null(&shift), spacetime_metric, inverse_spatial_metric);
+
+  auto& lapse = get<gr::Tags::Lapse<DataVector>>(computation_variables);
+  gr::lapse(make_not_null(&lapse), shift, spacetime_metric);
+
+  auto& dt_spacetime_metric = get<
+      ::Tags::dt<gr::Tags::SpacetimeMetric<3, ::Frame::Inertial, DataVector>>>(
+      computation_variables);
+  for (size_t a = 0; a < 4; ++a) {
```

Comment:
doesn't this function exist in PointwiseFunctions?  If not we should move it there...